<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Optimus v4.0.2 Methods | WARP</title>
    <meta name="generator" content="VuePress 1.7.1">
    <link rel="icon" href="/warp/logo.png">
    <meta name="description" content="WDL Analysis Research Pipelines">
    <meta name="theme-color" content="#3eaf7c">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <meta name="og:locale" content="en_US">
    
    <link rel="preload" href="/warp/assets/css/0.styles.9e4f499e.css" as="style"><link rel="preload" href="/warp/assets/js/app.a2b744f4.js" as="script"><link rel="preload" href="/warp/assets/js/2.3c82beca.js" as="script"><link rel="preload" href="/warp/assets/js/27.b0c93a9b.js" as="script"><link rel="prefetch" href="/warp/assets/js/10.c7b9fb28.js"><link rel="prefetch" href="/warp/assets/js/11.bd360b5c.js"><link rel="prefetch" href="/warp/assets/js/12.48f3249c.js"><link rel="prefetch" href="/warp/assets/js/13.b33887f2.js"><link rel="prefetch" href="/warp/assets/js/14.52da87ff.js"><link rel="prefetch" href="/warp/assets/js/15.41b0875b.js"><link rel="prefetch" href="/warp/assets/js/16.fcad7e22.js"><link rel="prefetch" href="/warp/assets/js/17.19040465.js"><link rel="prefetch" href="/warp/assets/js/18.5cd85943.js"><link rel="prefetch" href="/warp/assets/js/19.f71a8bf8.js"><link rel="prefetch" href="/warp/assets/js/20.417014aa.js"><link rel="prefetch" href="/warp/assets/js/21.ce8d83ec.js"><link rel="prefetch" href="/warp/assets/js/22.71cb5734.js"><link rel="prefetch" href="/warp/assets/js/23.c1051ff5.js"><link rel="prefetch" href="/warp/assets/js/24.bd99bf5b.js"><link rel="prefetch" href="/warp/assets/js/25.5d2d381c.js"><link rel="prefetch" href="/warp/assets/js/26.a4e1c4c3.js"><link rel="prefetch" href="/warp/assets/js/28.1d11009b.js"><link rel="prefetch" href="/warp/assets/js/29.3e8b5d07.js"><link rel="prefetch" href="/warp/assets/js/3.acd2e54e.js"><link rel="prefetch" href="/warp/assets/js/30.f1f00eaf.js"><link rel="prefetch" href="/warp/assets/js/31.0c140f0a.js"><link rel="prefetch" href="/warp/assets/js/32.1f8e28eb.js"><link rel="prefetch" href="/warp/assets/js/33.f503b968.js"><link rel="prefetch" href="/warp/assets/js/34.87764388.js"><link rel="prefetch" href="/warp/assets/js/35.44ee75eb.js"><link rel="prefetch" href="/warp/assets/js/36.487613dc.js"><link rel="prefetch" href="/warp/assets/js/4.a326751d.js"><link rel="prefetch" href="/warp/assets/js/5.7e7dcbe0.js"><link rel="prefetch" href="/warp/assets/js/6.de7f6e39.js"><link rel="prefetch" href="/warp/assets/js/7.d3b8d91c.js"><link rel="prefetch" href="/warp/assets/js/8.11f1f308.js"><link rel="prefetch" href="/warp/assets/js/9.6a43dcb9.js">
    <link rel="stylesheet" href="/warp/assets/css/0.styles.9e4f499e.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/warp/" class="home-link router-link-active"><!----> <span class="site-name">WARP</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="/warp/documentation/Intro.html" class="nav-link">
  Documentation
</a></div><div class="nav-item"><a href="/warp/blog/" class="nav-link">
  Blog
</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="Learn More" class="dropdown-title"><span class="title">Learn More</span> <span class="arrow down"></span></button> <button type="button" aria-label="Learn More" class="mobile-dropdown-title"><span class="title">Learn More</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><h4>
          Explore
        </h4> <ul class="dropdown-subitem-wrapper"><li class="dropdown-subitem"><a href="/warp/research/" class="nav-link">
  Research
</a></li></ul></li><li class="dropdown-item"><h4>
          Resources
        </h4> <ul class="dropdown-subitem-wrapper"><li class="dropdown-subitem"><a href="/warp/contribution/" class="nav-link">
  Contribution Guide
</a></li><li class="dropdown-subitem"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer" class="nav-link external">
  Changelog
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></li></ul></li></ul></div></div> <a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/warp/documentation/Intro.html" class="nav-link">
  Documentation
</a></div><div class="nav-item"><a href="/warp/blog/" class="nav-link">
  Blog
</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="Learn More" class="dropdown-title"><span class="title">Learn More</span> <span class="arrow down"></span></button> <button type="button" aria-label="Learn More" class="mobile-dropdown-title"><span class="title">Learn More</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><h4>
          Explore
        </h4> <ul class="dropdown-subitem-wrapper"><li class="dropdown-subitem"><a href="/warp/research/" class="nav-link">
  Research
</a></li></ul></li><li class="dropdown-item"><h4>
          Resources
        </h4> <ul class="dropdown-subitem-wrapper"><li class="dropdown-subitem"><a href="/warp/contribution/" class="nav-link">
  Contribution Guide
</a></li><li class="dropdown-subitem"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer" class="nav-link external">
  Changelog
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></li></ul></li></ul></div></div> <a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></nav>  <ul class="sidebar-links"><li><section class="sidebar-group depth-0"><p class="sidebar-heading"><span>Welcome to WARP</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/warp/documentation/Intro.html" class="sidebar-link">Welcome to WARP</a></li></ul></section></li><li><section class="sidebar-group depth-0"><p class="sidebar-heading"><span>About WARP</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/warp/documentation/About_WARP/BestPractices.html" class="sidebar-link">Best Practices for Building Data Processing Pipelines</a></li><li><a href="/warp/documentation/About_WARP/PipelineRequirements.html" class="sidebar-link">Pipeline Requirements</a></li><li><a href="/warp/documentation/About_WARP/TestingPipelines.html" class="sidebar-link">Testing Pipelines</a></li><li><a href="/warp/documentation/About_WARP/VersionAndReleasePipelines.html" class="sidebar-link">Version and Release Pipelines</a></li></ul></section></li><li><section class="sidebar-group depth-0"><p class="sidebar-heading open"><span>Pipelines</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading"><span>CEMBA MethylC Seq Pipeline</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading"><span>Exome Germline Single Sample Pipeline</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading"><span>Illumina Genotyping Arrays Pipeline</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading open"><span>Optimus Pipeline</span> <span class="arrow down"></span></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/warp/documentation/Pipelines/Optimus_Pipeline/" aria-current="page" class="sidebar-link">Optimus Overview</a></li><li><a href="/warp/documentation/Pipelines/Optimus_Pipeline/optimus.methods.html" aria-current="page" class="active sidebar-link">Optimus v4.0.2 Methods</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/warp/documentation/Pipelines/Optimus_Pipeline/optimus.methods.html#single-cell-sc-rna-mode" class="sidebar-link">Single-cell (sc_rna mode)</a></li><li class="sidebar-sub-header"><a href="/warp/documentation/Pipelines/Optimus_Pipeline/optimus.methods.html#single-nuclei-sn-rna-mode" class="sidebar-link">Single-nuclei (sn_rna mode)</a></li></ul></li><li><a href="/warp/documentation/Pipelines/Optimus_Pipeline/Bam_tags.html" class="sidebar-link">What tags are included in an Optimus BAM file?</a></li><li><a href="/warp/documentation/Pipelines/Optimus_Pipeline/Loom_schema.html" class="sidebar-link">What's in the Optimus Pipeline Loom File?</a></li></ul></section></li><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading"><span>Single Cell ATAC Seq Pipeline</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading"><span>Smart-seq2 Multi Sample Pipeline</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading"><span>Smart-seq2 Single Sample Pipeline</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading"><span>Whole Genome Germline Single Sample Pipeline</span> <span class="arrow right"></span></p> <!----></section></li></ul></section></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="optimus-v4-0-2-methods"><a href="#optimus-v4-0-2-methods" class="header-anchor">#</a> Optimus v4.0.2 Methods</h1> <p>Below we provide a sample methods sections for a publication, separated into single-cell or single-nuclei use cases. For the complete pipeline documentation, see the <a href="/warp/documentation/Pipelines/Optimus_Pipeline/" class="router-link-active">Optimus</a>.</p> <h1 id="methods"><a href="#methods" class="header-anchor">#</a> Methods</h1> <h2 id="single-cell-sc-rna-mode"><a href="#single-cell-sc-rna-mode" class="header-anchor">#</a> Single-cell (sc_rna mode)</h2> <p>Data preprocessing and count matrix construction were performed using the Optimus v4.0.2 Pipeline. Briefly, FASTQ files were converted to unaligned BAM (uBMA) using Picard v2.10.10 and reads were appended with raw UMI and corrected cell barcode sequences using Single Cell Tools (sctools) v0.3.10 and the 10x Genomics barcodes whitelist, allowing for up to one edit distance (Levenshtein distance).</p> <p>uBAMs were then aligned to GENCODE mouse (M21) or human (V27) references using STAR v2.5.3a with default parameters in addition to</p> <div class="language- extra-class"><pre class="language-text"><code>--BAM unsorted --outSAMattributes all --outSAMunmapped --readFilestype SAM SE
</code></pre></div><p>Genes were annotated and reads were tagged with Drop-seq Tools v1.12 using the TagReadwithGeneExon function.</p> <p>UMIs were then corrected and duplicate reads marked using UMI-tools v0.0.1 with default parameters in addition to</p> <div class="language- extra-class"><pre class="language-text"><code>--extract-umi-method=tag --umi-tag UR --cell-tag CB --gene-tag GE --umi-group-tag UB --per-gene --per-cell --no-sort-output
</code></pre></div><p>All reads (UMI-corrected, duplicate, and untagged) were merged into a single BAM file and tagged. Gene and cell-specific metrics were calculated using the sctools v0.3.7 functions <code>CalculateGeneMetrics</code> and <code>CalculateCellMetrics</code>.</p> <p>Empty droplets were identified, but not removed to enable downstream filtering, using the DropletUtils v.1.2.1 with</p> <div class="language- extra-class"><pre class="language-text"><code>--fdr-cutoff 0.01 --emptydrops-niters 10000 --min-molecules 100 --emptydrops-lower 100
</code></pre></div><p>UMI-aware count matrices for exon-only alignments were produced using the sctools v0.3.7.
All cell and gene metrics (alignment, mitochondrial, and other QC metrics), count matrices and DropletUtils results were then aggregated into a final Loom file for downstream processing. The final outputs included the unfiltered Loom and unfiltered (but tagged) BAM files.</p> <p>An example of the pipeline and outputs is available on the Terra HCA Optimus Pipeline Featured Workspace (https://app.terra.bio/#workspaces/featured-workspaces-hca/HCA_Optimus_Pipeline), and additional documentation is available on GitHub (https://github.com/HumanCellAtlas/skylab/blob/master/pipelines/optimus/README.md). Examples of genomic references, whitelists, and other inputs are available in the Skylab repository (see example JSONs).</p> <h2 id="single-nuclei-sn-rna-mode"><a href="#single-nuclei-sn-rna-mode" class="header-anchor">#</a> Single-nuclei (sn_rna mode)</h2> <p>Data preprocessing and count matrix construction were performed using the Optimus v4.0.2 Pipeline. Briefly, FASTQ files were converted to unaligned BAM (uBMA) using Picard v2.10.10 and reads were appended with raw UMI and corrected cell barcode sequences using Single Cell Tools (sctools) v0.3.10 and the 10x Genomics barcodes whitelist, allowing for up to one edit distance (Levenshtein distance).</p> <p>uBAMs were aligned to GENCODE mouse (M21) or human (V27) references using STAR v2.5.3a with default parameters in addition to</p> <div class="language- extra-class"><pre class="language-text"><code>--BAM unsorted --outSAMattributes all --outSAMunmapped --readFilestype SAM SE
</code></pre></div><p>Genes were annotated and reads were tagged with Drop-seq Tools v2.3.0 using TagReadWithGeneFunction.</p> <p>UMIs were then corrected using UMI-tools v0.0.1 with default parameters in addition to</p> <div class="language- extra-class"><pre class="language-text"><code>--extract-umi-method=tag --umi-tag UR --cell-tag CB --gene-tag GE --umi-group-tag UB --per-gene --per-cell --no-sort-output
</code></pre></div><p>All reads (UMI-corrected, duplicate, and untagged) were merged into a single BAM file. Gene and cell-specific metrics were calculated using the sctools v.0.3.7 functions <code>CalculateGeneMetrics</code> and <code>CalculateCellMetrics</code>. UMI-aware count matrices for all alignments (introns, exons, UTRs) were produced using the sctools v0.3.7.</p> <p>All cell and gene metrics (alignment, mitochondrial, and other QC metrics), annotations, and count matrices were aggregated into a final Loom file for downstream processing. The final outputs included the unfiltered Loom and unfiltered (but tagged) BAM files.</p> <p>An example of the pipeline and outputs is available on the <a href="https://app.terra.bio/#workspaces/featured-workspaces-hca/HCA_Optimus_Pipeline" target="_blank" rel="noopener noreferrer">Terra HCA Optimus Pipeline Featured Workspace<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>, and additional documentation is available on <a href="https://github.com/HumanCellAtlas/skylab/blob/master/pipelines/optimus/README.md" target="_blank" rel="noopener noreferrer">GitHub<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>. Examples of genomic references, whitelists, and other inputs are available in the Skylab repository (see the *_example.json files at <a href="https://github.com/HumanCellAtlas/skylab/tree/master/pipelines/optimus" target="_blank" rel="noopener noreferrer">here<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>.</p></div> <footer class="page-edit"><div class="edit-link"><a href="https://github.com/broadinstitute/warp/edit/master/documentation/Pipelines/Optimus_Pipeline/optimus.methods.md" target="_blank" rel="noopener noreferrer">Edit this page</a> <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></div> <div class="last-updated"><span class="prefix">Last Updated:</span> <span class="time">10/22/2020, 12:11:42 AM</span></div></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/warp/documentation/Pipelines/Optimus_Pipeline/" class="prev router-link-active">
        Optimus Overview
      </a></span> <span class="next"><a href="/warp/documentation/Pipelines/Optimus_Pipeline/Bam_tags.html">
        What tags are included in an Optimus BAM file?
      </a>
      →
    </span></p></div> </main></div><div class="global-ui"><!----></div></div>
    <script src="/warp/assets/js/app.a2b744f4.js" defer></script><script src="/warp/assets/js/2.3c82beca.js" defer></script><script src="/warp/assets/js/27.b0c93a9b.js" defer></script>
  </body>
</html>
