(window.webpackJsonp=window.webpackJsonp||[]).push([[23],{399:function(e,t,r){"use strict";r.r(t);var n=r(19),o=Object(n.a)({},(function(){var e=this,t=e.$createElement,r=e._self._c||t;return r("ContentSlotsDistributor",{attrs:{"slot-key":e.$parent.slotKey}},[r("h1",{attrs:{id:"what-tags-are-included-in-an-optimus-bam-file"}},[r("a",{staticClass:"header-anchor",attrs:{href:"#what-tags-are-included-in-an-optimus-bam-file"}},[e._v("#")]),e._v(" What tags are included in an Optimus BAM file?")]),e._v(" "),r("p",[e._v("The Optimus Pipeline outputs a barcoded BAM file of aligned reads. There are multiple tags within the BAM file, including standard tags from "),r("a",{attrs:{href:"https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/bam",target:"_blank",rel:"noopener noreferrer"}},[e._v("10X genomics"),r("OutboundLink")],1),e._v(" and "),r("a",{attrs:{href:"https://samtools.github.io/hts-specs/SAMtags.pdf",target:"_blank",rel:"noopener noreferrer"}},[e._v("Sequence Alignment Map (SAM) files"),r("OutboundLink")],1),e._v(". The table below details the tags used by the Optimus Pipeline and the relevant sources/tools from which the pipeline obtains the tags.")]),e._v(" "),r("table",[r("thead",[r("tr",[r("th",[e._v("Optimus Pipeline BAM Tag")]),e._v(" "),r("th",[e._v("Details")]),e._v(" "),r("th",[e._v("Source")])])]),e._v(" "),r("tbody",[r("tr",[r("td",[e._v("AS")]),e._v(" "),r("td",[e._v("Alignment score generated by aligner")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://samtools.github.io/hts-specs/SAMtags.pdf",target:"_blank",rel:"noopener noreferrer"}},[e._v("SAM"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("CB")]),e._v(" "),r("td",[e._v("Chromium cellular barcode sequence that is error-corrected and confirmed against a list of known-good barcode sequences")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/bam",target:"_blank",rel:"noopener noreferrer"}},[e._v("10X Genomics"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("CR")]),e._v(" "),r("td",[e._v("Chromium cellular barcode sequence as reported by the sequencer")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/bam",target:"_blank",rel:"noopener noreferrer"}},[e._v("10X Genomics"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("CY")]),e._v(" "),r("td",[e._v("Chromium cellular barcode read quality. Phred scores as reported by sequencer")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/bam",target:"_blank",rel:"noopener noreferrer"}},[e._v("10X Genomics"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("GE")]),e._v(" "),r("td",[e._v("Gene exon")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://github.com/broadinstitute/Drop-seq",target:"_blank",rel:"noopener noreferrer"}},[e._v("Drop-seq Tools"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("GS")]),e._v(" "),r("td",[e._v("Reserved for backwards compatibility reasons")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://samtools.github.io/hts-specs/SAMtags.pdf",target:"_blank",rel:"noopener noreferrer"}},[e._v("SAM"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("HI")]),e._v(" "),r("td",[e._v("Query hit index")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://samtools.github.io/hts-specs/SAMtags.pdf",target:"_blank",rel:"noopener noreferrer"}},[e._v("SAM"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("MD")]),e._v(" "),r("td",[e._v("String for mismatching indexes")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://samtools.github.io/hts-specs/SAMtags.pdf",target:"_blank",rel:"noopener noreferrer"}},[e._v("SAM"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("NH")]),e._v(" "),r("td",[e._v("Number of reported alignments that contain the query in the current record")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://samtools.github.io/hts-specs/SAMtags.pdf",target:"_blank",rel:"noopener noreferrer"}},[e._v("SAM"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("NM")]),e._v(" "),r("td",[e._v("Edit difference to the reference")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://samtools.github.io/hts-specs/SAMtags.pdf",target:"_blank",rel:"noopener noreferrer"}},[e._v("SAM"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("RG")]),e._v(" "),r("td",[e._v("Read group")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://samtools.github.io/hts-specs/SAMtags.pdf",target:"_blank",rel:"noopener noreferrer"}},[e._v("SAM"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("UB")]),e._v(" "),r("td",[e._v("Chromium molecular barcode sequence that is error-corrected among other molecular barcodes with the same cellular barcode and gene alignment")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/bam",target:"_blank",rel:"noopener noreferrer"}},[e._v("10X Genomics"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("UG")]),e._v(" "),r("td",[e._v("Read group ID")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://github.com/CGATOxford/UMI-tools",target:"_blank",rel:"noopener noreferrer"}},[e._v("Umi-tools"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("UR")]),e._v(" "),r("td",[e._v("Chromium molecular barcode sequence as reported by the sequencer")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/bam",target:"_blank",rel:"noopener noreferrer"}},[e._v("10X Genomics"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("UY")]),e._v(" "),r("td",[e._v("Chromium molecular barcode read quality. Phred scores as reported by sequencer")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/bam",target:"_blank",rel:"noopener noreferrer"}},[e._v("10X Genomics"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("XF")]),e._v(" "),r("td",[e._v("Extra alignment flags")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/bam",target:"_blank",rel:"noopener noreferrer"}},[e._v("10X Genomics"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("jI")]),e._v(" "),r("td",[e._v("[jI:B:I,Start1,End1,Start2,End2,...] Start and End of introns for all junctions (1-based)")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://github.com/alexdobin/STAR",target:"_blank",rel:"noopener noreferrer"}},[e._v("STAR"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("jM")]),e._v(" "),r("td",[e._v("[jM:B:c,M1,M2,...] Intron motifs for all junctions (i.e. N in CIGAR)")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://github.com/alexdobin/STAR",target:"_blank",rel:"noopener noreferrer"}},[e._v("STAR"),r("OutboundLink")],1)])]),e._v(" "),r("tr",[r("td",[e._v("nM")]),e._v(" "),r("td",[e._v("The number of mismatches per (paired) alignment")]),e._v(" "),r("td",[r("a",{attrs:{href:"https://github.com/alexdobin/STAR",target:"_blank",rel:"noopener noreferrer"}},[e._v("STAR"),r("OutboundLink")],1)])])])])])}),[],!1,null,null,null);t.default=o.exports}}]);