(window.webpackJsonp=window.webpackJsonp||[]).push([[24],{391:function(e,t,r){"use strict";r.r(t);var a=r(25),n=Object(a.a)({},(function(){var e=this,t=e.$createElement,r=e._self._c||t;return r("ContentSlotsDistributor",{attrs:{"slot-key":e.$parent.slotKey}},[r("h1",{attrs:{id:"introduction-to-the-exome-germline-single-sample-pipeline"}},[r("a",{staticClass:"header-anchor",attrs:{href:"#introduction-to-the-exome-germline-single-sample-pipeline"}},[e._v("#")]),e._v(" Introduction to the Exome Germline Single Sample Pipeline")]),e._v(" "),r("table",[r("thead",[r("tr",[r("th",{staticStyle:{"text-align":"center"}},[e._v("Pipeline Version")]),e._v(" "),r("th",{staticStyle:{"text-align":"center"}},[e._v("Date Updated")]),e._v(" "),r("th",{staticStyle:{"text-align":"center"}},[e._v("Documentation Author")]),e._v(" "),r("th",{staticStyle:{"text-align":"center"}},[e._v("Questions or Feedback")])])]),e._v(" "),r("tbody",[r("tr",[r("td",{staticStyle:{"text-align":"center"}},[r("a",{attrs:{href:"https://github.com/broadinstitute/warp/releases",target:"_blank",rel:"noopener noreferrer"}},[e._v("ExomeGermlineSingleSample_v2.0"),r("OutboundLink")],1)]),e._v(" "),r("td",{staticStyle:{"text-align":"center"}},[e._v("June 10, 2020")]),e._v(" "),r("td",{staticStyle:{"text-align":"center"}},[r("a",{attrs:{href:"mailto:ekiernan@broadinstitute.org"}},[e._v("Elizabeth Kiernan")])]),e._v(" "),r("td",{staticStyle:{"text-align":"center"}},[e._v("Please file GitHub issues in dsde-pipelines or contact "),r("a",{attrs:{href:"mailto:kdegatano@broadinstitute.org"}},[e._v("Kylee Degatano")])])])])]),e._v(" "),r("h2",{attrs:{id:"table-of-contents"}},[r("a",{staticClass:"header-anchor",attrs:{href:"#table-of-contents"}},[e._v("#")]),e._v(" Table of Contents")]),e._v(" "),r("ul",[r("li",[r("a",{attrs:{href:"#introduction-to-the-exome-germline-single-sample-pipeline"}},[e._v("Introduction to the Exome Germline Single Sample Pipeline")])]),e._v(" "),r("li",[r("a",{attrs:{href:"#set-up"}},[e._v("Set-up")]),e._v(" "),r("ul",[r("li",[r("a",{attrs:{href:"#workflow-installation-and-requirements"}},[e._v("Workflow Installation and Requirements")])]),e._v(" "),r("li",[r("a",{attrs:{href:"#input-requirements-and-expectations"}},[e._v("Input Requirements and Expectations")])])])]),e._v(" "),r("li",[r("a",{attrs:{href:"#workflow-tasks-and-tools"}},[e._v("Workflow Tasks and Tools")])]),e._v(" "),r("li",[r("a",{attrs:{href:"#workflow-outputs"}},[e._v("Workflow Outputs")])]),e._v(" "),r("li",[r("a",{attrs:{href:"#important-notes"}},[e._v("Important Notes")])]),e._v(" "),r("li",[r("a",{attrs:{href:"#contact-us"}},[e._v("Contact Us")])]),e._v(" "),r("li",[r("a",{attrs:{href:"#licensing"}},[e._v("Licensing")])])]),e._v(" "),r("p",[e._v("The Exome Germline Single Sample pipeline implements data pre-processing and initial variant calling according to the GATK Best Practices for germline SNP and Indel discovery in human exome sequencing data. For a broad overview of the pipeline processes, read the GATK Best Practices documentation for "),r("a",{attrs:{href:"https://gatk.broadinstitute.org/hc/en-us/articles/360035535912",target:"_blank",rel:"noopener noreferrer"}},[e._v("data pre-processing"),r("OutboundLink")],1),e._v(" and for "),r("a",{attrs:{href:"https://gatk.broadinstitute.org/hc/en-us/articles/360035535932",target:"_blank",rel:"noopener noreferrer"}},[e._v("germline short variant discovery"),r("OutboundLink")],1),e._v(".")]),e._v(" "),r("h2",{attrs:{id:"set-up"}},[r("a",{staticClass:"header-anchor",attrs:{href:"#set-up"}},[e._v("#")]),e._v(" Set-up")]),e._v(" "),r("h3",{attrs:{id:"workflow-installation-and-requirements"}},[r("a",{staticClass:"header-anchor",attrs:{href:"#workflow-installation-and-requirements"}},[e._v("#")]),e._v(" Workflow Installation and Requirements")]),e._v(" "),r("p",[e._v("The Exome Germline Single Sample workflow is written in the Workflow Description Language WDL and can be downloaded by cloning the GitHub repository dsde-pipelines. The workflow can be deployed using "),r("a",{attrs:{href:"https://github.com/broadinstitute/cromwell",target:"_blank",rel:"noopener noreferrer"}},[e._v("Cromwell"),r("OutboundLink")],1),e._v(", a GA4GH compliant, flexible workflow management system that supports multiple computing platforms. For the latest workflow version and release notes, please see the Exome Germline Single Sample "),r("RouterLink",{attrs:{to:"/documentation/Pipelines/Exome_Germline_Single_Sample_Pipeline/ExomeGermlineSingleSample.changelog.html"}},[e._v("changelog")]),e._v(".")],1),e._v(" "),r("h3",{attrs:{id:"input-requirements-and-expectations"}},[r("a",{staticClass:"header-anchor",attrs:{href:"#input-requirements-and-expectations"}},[e._v("#")]),e._v(" Input Requirements and Expectations")]),e._v(" "),r("ul",[r("li",[e._v("Human exome sequencing data in unmapped BAM (uBAM) format")]),e._v(" "),r("li",[e._v("One or more read groups, one per uBAM file, all belonging to a single sample (SM)")]),e._v(" "),r("li",[e._v("Input uBAM files must additionally comply with the following requirements:\n"),r("ul",[r("li",[e._v('Filenames all have the same suffix (we use ".unmapped.bam")')]),e._v(" "),r("li",[e._v("Files must pass validation by ValidateSamFile")]),e._v(" "),r("li",[e._v("Reads are provided in query-sorted order")]),e._v(" "),r("li",[e._v("All reads must have an RG tag")])])]),e._v(" "),r("li",[e._v('GVCF output names must end in ".g.vcf.gz"')]),e._v(" "),r("li",[e._v("Reference genome must be Hg38 with ALT contigs")]),e._v(" "),r("li",[e._v("Unique exome calling, target, and bait "),r("a",{attrs:{href:"https://gatk.broadinstitute.org/hc/en-us/articles/360035531852",target:"_blank",rel:"noopener noreferrer"}},[e._v(".interval_list"),r("OutboundLink")],1),e._v(" obtained from the sequencing provider. Generally the calling, target, and bait files will not be the same.")])]),e._v(" "),r("h2",{attrs:{id:"workflow-tasks-and-tools"}},[r("a",{staticClass:"header-anchor",attrs:{href:"#workflow-tasks-and-tools"}},[e._v("#")]),e._v(" Workflow Tasks and Tools")]),e._v(" "),r("p",[e._v("The Exome Germline Single Sample "),r("a",{attrs:{href:"ExomeGermlineSingleSample.wdl"}},[e._v("workflow")]),e._v(" imports a series of tasks from the dsde-pipelines "),r("RouterLink",{attrs:{to:"/tasks/broad/"}},[e._v("tasks library")]),e._v(" and a DNASeq struct ("),r("a",{attrs:{href:"../../../../../../structs/dna_seq/DNASeqStructs.wdl"}},[e._v("DNASeqStructs.wdl")]),e._v(") containing reference files from the "),r("RouterLink",{attrs:{to:"/structs/"}},[e._v("structs library")]),e._v(".")],1),e._v(" "),r("p",[e._v("You can read more about the software tools implemented in these tasks by reading the GATK "),r("a",{attrs:{href:"https://gatk.broadinstitute.org/hc/en-us/articles/360035535912",target:"_blank",rel:"noopener noreferrer"}},[e._v("data pre-processing"),r("OutboundLink")],1),e._v(" and "),r("a",{attrs:{href:"https://gatk.broadinstitute.org/hc/en-us/articles/360035535932",target:"_blank",rel:"noopener noreferrer"}},[e._v("germline short variant discovery"),r("OutboundLink")],1),e._v(" documentation.")]),e._v(" "),r("h2",{attrs:{id:"workflow-outputs"}},[r("a",{staticClass:"header-anchor",attrs:{href:"#workflow-outputs"}},[e._v("#")]),e._v(" Workflow Outputs")]),e._v(" "),r("ul",[r("li",[e._v("CRAM, CRAM index, and CRAM MD5")]),e._v(" "),r("li",[e._v("GVCF and its GVCF index")]),e._v(" "),r("li",[e._v("BQSR report")]),e._v(" "),r("li",[e._v("Summary metrics; to read more about any particular metric, you can search the metric using the "),r("a",{attrs:{href:"https://gatk.broadinstitute.org/hc/en-us/categories/360002302312",target:"_blank",rel:"noopener noreferrer"}},[e._v("GATK documentation search"),r("OutboundLink")],1)])]),e._v(" "),r("h2",{attrs:{id:"important-notes"}},[r("a",{staticClass:"header-anchor",attrs:{href:"#important-notes"}},[e._v("#")]),e._v(" Important Notes")]),e._v(" "),r("ul",[r("li",[e._v("Runtime parameters are optimized for Broad's Google Cloud Platform implementation.")]),e._v(" "),r("li",[e._v("For help running workflows on the Google Cloud Platform or locally please\nview the following tutorial "),r("a",{attrs:{href:"https://gatk.broadinstitute.org/hc/en-us/articles/360035530952",target:"_blank",rel:"noopener noreferrer"}},[e._v("(How to) Execute Workflows from the gatk-workflows Git Organization"),r("OutboundLink")],1),e._v(".")]),e._v(" "),r("li",[e._v("Please visit the "),r("a",{attrs:{href:"https://gatk.broadinstitute.org/hc/en-us/categories/360002310591",target:"_blank",rel:"noopener noreferrer"}},[e._v("GATK Technical Documentation"),r("OutboundLink")],1),e._v(" site for further documentation on our workflows and tools.")])]),e._v(" "),r("h2",{attrs:{id:"contact-us"}},[r("a",{staticClass:"header-anchor",attrs:{href:"#contact-us"}},[e._v("#")]),e._v(" Contact Us")]),e._v(" "),r("p",[e._v("This material is provided by the Data Science Platform group at the Broad Institute. Please direct any questions or concerns to one of our forum sites : "),r("a",{attrs:{href:"https://gatk.broadinstitute.org/hc/en-us/community/topics",target:"_blank",rel:"noopener noreferrer"}},[e._v("GATK"),r("OutboundLink")],1),e._v(" or "),r("a",{attrs:{href:"https://support.terra.bio/hc/en-us/community/topics/360000500432",target:"_blank",rel:"noopener noreferrer"}},[e._v("Terra"),r("OutboundLink")],1),e._v(".")]),e._v(" "),r("h2",{attrs:{id:"licensing"}},[r("a",{staticClass:"header-anchor",attrs:{href:"#licensing"}},[e._v("#")]),e._v(" Licensing")]),e._v(" "),r("p",[e._v("Copyright Broad Institute, 2020 | BSD-3")]),e._v(" "),r("p",[e._v("The workflow script is released under the "),r("strong",[e._v("WDL open source code license (BSD-3)")]),e._v(" (full license text at https://github.com/openwdl/wdl/blob/master/LICENSE). However, please note that the programs it calls may be subject to different licenses. Users are responsible for checking that they are authorized to run all programs before running this script.")])])}),[],!1,null,null,null);t.default=n.exports}}]);