<!doctype html>
<html lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.1">
<link rel="alternate" type="application/rss+xml" href="/warp/blog/rss.xml" title="WARP Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/warp/blog/atom.xml" title="WARP Blog Atom Feed">
<link rel="preconnect" href="https://www.google-analytics.com">
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-181048107-1","auto"),ga("set","anonymizeIp",!0),ga("send","pageview")</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
<link rel="search" type="application/opensearchdescription+xml" title="WARP" href="/warp/opensearch.xml">
<script>!function(t,h,e,j,s,n){t.hj=t.hj||function(){(t.hj.q=t.hj.q||[]).push(arguments)},t._hjSettings={hjid:2427684,hjsv:6},s=h.getElementsByTagName("head")[0],(n=h.createElement("script")).async=1,n.src="https://static.hotjar.com/c/hotjar-"+t._hjSettings.hjid+".js?sv="+t._hjSettings.hjsv,s.appendChild(n)}(window,document)</script><title data-react-helmet="true">Smart-seq2 Single Nucleus Overview | WARP</title><meta data-react-helmet="true" property="og:url" content="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Pipeline/README"><meta data-react-helmet="true" name="docsearch:language" content="en"><meta data-react-helmet="true" name="docsearch:version" content="current"><meta data-react-helmet="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Smart-seq2 Single Nucleus Overview | WARP"><meta data-react-helmet="true" name="description" content="| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |"><meta data-react-helmet="true" property="og:description" content="| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |"><link data-react-helmet="true" rel="shortcut icon" href="/warp/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Pipeline/README"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Pipeline/README" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Pipeline/README" hreflang="x-default"><link data-react-helmet="true" rel="preconnect" href="https://BH4D9OD16A-dsn.algolia.net" crossorigin="anonymous"><link rel="stylesheet" href="/warp/assets/css/styles.e4a54f34.css">
<link rel="preload" href="/warp/assets/js/runtime~main.6a48d3f3.js" as="script">
<link rel="preload" href="/warp/assets/js/main.41bb09f5.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#main" class="skipToContent_1oUP shadow--md">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/warp/"><b class="navbar__title">WARP Pipelines</b></a></div><div class="navbar__items navbar__items--right"><a class="navbar__item navbar__link navbar__link--active" href="/warp/docs/get-started">Documentation</a><a class="navbar__item navbar__link" href="/warp/blog">Blog</a><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link" aria-label="GitHub repository"></a><div class="react-toggle displayOnlyInLargeViewport_GrZ2 react-toggle--disabled"><div class="react-toggle-track" role="button" tabindex="-1"><div class="react-toggle-track-check"><span class="toggle_71bT">🌜</span></div><div class="react-toggle-track-x"><span class="toggle_71bT">🌞</span></div><div class="react-toggle-thumb"></div></div><input type="checkbox" class="react-toggle-screenreader-only" aria-label="Switch between dark and light mode"></div><div class="searchBox_1Doo"><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><span class="DocSearch-Button-Container"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">Search</span></span></button></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div><div class="navbar-sidebar"><div class="navbar-sidebar__brand"><a class="navbar__brand" href="/warp/"><b class="navbar__title">WARP Pipelines</b></a></div><div class="navbar-sidebar__items"><div class="menu"><ul class="menu__list"><li class="menu__list-item"><a class="menu__link navbar__link--active" href="/warp/docs/get-started">Documentation</a></li><li class="menu__list-item"><a class="menu__link" href="/warp/blog">Blog</a></li><li class="menu__list-item"><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="menu__link header-github-link" aria-label="GitHub repository"></a></li></ul></div></div></div></nav><div class="main-wrapper docs-wrapper doc-page"><div class="docPage_31aa"><aside class="docSidebarContainer_3Kbt"><div class="sidebar_15mo"><nav class="menu menu--responsive thin-scrollbar menu_Bmed" aria-label="Sidebar navigation"><button aria-label="Open menu" aria-haspopup="true" class="button button--secondary button--sm menu__button" type="button"><svg class="sidebarMenuIcon_fgN0" width="24" height="24" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/warp/docs/get-started">Welcome to WARP</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">About WARP</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/About_WARP/BestPractices">Best Practices for Building Data Processing Pipelines</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/About_WARP/PipelineRequirements">Pipeline Requirements</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/About_WARP/TestingPipelines">Testing Pipelines</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/About_WARP/VersionAndReleasePipelines">Version and Release Pipelines</a></li></ul></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#!">WARP Pipelines</a><ul class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">CEMBA</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/CEMBA_MethylC_Seq_Pipeline/README">CEMBA Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/CEMBA_MethylC_Seq_Pipeline/CEMBA.methods">CEMBA_v1.1.0 Publication Methods</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">Exome Germline Single Sample</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Exome_Germline_Single_Sample_Pipeline/README">Exome Germline Single Sample Overview</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">GDC Whole Genome Somatic Single Sample</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Genomic_Data_Commons_Whole_Genome_Somatic/README">Genomic Data Commons (GDC) Whole Genome Somatic Single Sample Overview</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">Illumina Genotyping Array</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Illumina_Genotyping_Arrays_Pipeline/IlluminaGenotypingArray.documentation">Illumina Genotyping Array Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Illumina_Genotyping_Arrays_Pipeline/Illumina_genotyping_array_spec">VCF Overview: Illumina Genotyping Array</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">Optimus</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Optimus_Pipeline/README">Optimus Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Optimus_Pipeline/optimus.methods">Optimus v4.2.4 Methods</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Optimus_Pipeline/Bam_tags">What tags are included in an Optimus BAM file?</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Optimus_Pipeline/Loom_schema">Optimus Count Matrix Overview</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">Single Cell ATAC</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Single_Cell_ATAC_Seq_Pipeline/README">scATAC Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Single_Cell_ATAC_Seq_Pipeline/scatac.methods">scATAC v1.2.0 Methods</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">Smart-seq2 Multi-Sample</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/README">Smart-seq2 Multi-Sample Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/smart-seq2.methods">Smart-seq2 Multi-Sample v2.2.1 Publication Methods</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/Loom_schema">What&#x27;s in the Smart-seq2 Multi Sample Pipeline Loom File?</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">Smart-seq2 Single Nucleus Multi-Sample</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/README">Smart-seq2 Single Nucleus Multi-Sample Overview</a></li></ul></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#!" tabindex="0">Smart-seq2 Single Nucleus</a><ul class="menu__list"><li class="menu__list-item"><a aria-current="page" class="menu__link menu__link--active active" tabindex="0" href="/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Pipeline/README">Smart-seq2 Single Nucleus Overview</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">Smart-seq2 Single Sample</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/README">Smart-seq2 Single Sample Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/Loom_schema">What&#x27;s in the Smart-seq2 Single Sample Pipeline Loom File?</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">Whole Genome Germline Single Sample</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Whole_Genome_Germline_Single_Sample_Pipeline/README">Whole Genome Germline Single Sample Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Whole_Genome_Germline_Single_Sample_Pipeline/wgs.methods">Whole Genome Germline Single Sample v2.2.0 Methods</a></li></ul></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Contribution Guide</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/contribution/README">Welcome</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="-1">Contribute to WARP</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/contribution/contribute_to_warp/changelog_style">Changelog Style Guide Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/contribution/contribute_to_warp/contribution-guidelines">Contributing to Existing Pipelines</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/contribution/contribute_to_warp/contribution-tutorial">Example Contribution</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="-1">Contribute to WARP Documentation</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/contribution/contribute_to_warp_docs/docsite_maintenance">WARP Documentation Maintenance Guide</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/contribution/contribute_to_warp_docs/doc_style">Documentation Style Guide</a></li></ul></li></ul></li></ul></nav></div></aside><main class="docMainContainer_3ufF"><div class="container padding-top--md padding-bottom--lg docItemWrapper_3FMP"><div class="row"><div class="col docItemCol_3FnS"><div class="docItemContainer_33ec"><article><div class="markdown"><header><h1 class="h1Heading_27L5">Smart-seq2 Single Nucleus Overview</h1></header><table><thead><tr><th align="center">Pipeline Version</th><th align="center">Date Updated</th><th align="center">Documentation Author</th><th align="center">Questions or Feedback</th></tr></thead><tbody><tr><td align="center"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer">SmartSeq2SingleNucleus_v1.0.0</a></td><td align="center">May, 2021</td><td align="center"><a href="mailto:ekiernan@broadinstitute.org" target="_blank" rel="noopener noreferrer">Elizabeth Kiernan</a></td><td align="center">Please file GitHub issues in WARP or contact <a href="mailto:kdegatano@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kylee Degatano</a></td></tr></tbody></table><p><img src="/warp/assets/images/snSS2-2ed3badac7ac6bcd5ac8328893174814.png"></p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="introduction-to-the-smart-seq2-single-nucleus-pipeline"></a>Introduction to the Smart-seq2 Single Nucleus Pipeline<a class="hash-link" href="#introduction-to-the-smart-seq2-single-nucleus-pipeline" title="Direct link to heading">#</a></h2><p>The Smart-seq2 Single Nucleus (snSS2) pipeline was developed in collaboration with the <a href="https://biccn.org/" target="_blank" rel="noopener noreferrer">BRAIN Initiative Cell Census Network</a> (BICCN) to process single-nucleus RNAseq (snRNAseq) data generated by <a href="https://www.nature.com/articles/nmeth.2639" target="_blank" rel="noopener noreferrer">Smart-seq2 assays</a>. The pipeline&#x27;s workflow is written in WDL, is freely available in the <a href="https://github.com/broadinstitute/warp/tree/develop/pipelines/skylab/smartseq2_single_nucleus/SmartSeq2SingleNucleus.wdl" target="_blank" rel="noopener noreferrer">WARP repository</a> on GitHub, and can be run by any compliant WDL runner (e.g. <a href="https://github.com/broadinstitute/cromwell" target="_blank" rel="noopener noreferrer">Crowmell</a>). </p><p>The pipeline is designed to process snRNA-seq data from an individual cell (to process multiple cells, see the <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_nucleus_multisample/MultiSampleSmartSeq2SingleNucleus.wdl" target="_blank" rel="noopener noreferrer">Smart-seq2 Single Nucleus Multi-Sample</a> Pipeline).</p><p>Overall, the snSS2 workflow trims paired-end FASTQ files, aligns reads to the genome using a modified GTF, counts intronic and exonic reads, and calculates quality control metrics.</p><p>The pipeline has been scientifically validated by the BRAIN Institute. Read more in the <a href="#validation">validation section</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="quick-start-table"></a>Quick Start Table<a class="hash-link" href="#quick-start-table" title="Direct link to heading">#</a></h2><table><thead><tr><th>Pipeline Features</th><th>Description</th><th>Source</th></tr></thead><tbody><tr><td>Assay Type</td><td>Smart-seq2 Single Nucleus</td><td><a href="https://www.nature.com/articles/nprot.2014.006" target="_blank" rel="noopener noreferrer">Smart-seq2</a></td></tr><tr><td>Overall Workflow</td><td>Quality control and transcriptome quantification</td><td>Code available from the <a href="https://github.com/broadinstitute/warp/tree/develop/pipelines/skylab/smartseq2_single_nucleus/SmartSeq2SingleNucleus.wdl" target="_blank" rel="noopener noreferrer">WARP repository</a> in GitHub</td></tr><tr><td>Workflow Language</td><td>WDL</td><td><a href="https://github.com/openwdl/wdl" target="_blank" rel="noopener noreferrer">openWDL</a></td></tr><tr><td>Genomic Reference Sequence (for validation)</td><td>GRCm38 mouse genome primary sequence</td><td>GENCODE GRCm38 <a href="http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M23/GRCm38.primary_assembly.genome.fa.gz" target="_blank" rel="noopener noreferrer">Mouse</a></td></tr><tr><td>Transcriptomic Reference Annotation (for validation)</td><td>Modified <a href="https://www.gencodegenes.org/mouse/release_M23.html" target="_blank" rel="noopener noreferrer">M23</a> GTF built with the  <a href="https://github.com/broadinstitute/warp/tree/master/pipelines/skylab/build_indices/BuildIndices.wdl" target="_blank" rel="noopener noreferrer">BuildIndices workflow</a></td><td>GENCODE</td></tr><tr><td>Aligner</td><td>STAR (v.2.7.8a)</td><td><a href="https://github.com/alexdobin/STAR" target="_blank" rel="noopener noreferrer">STAR</a></td></tr><tr><td>QC Metrics</td><td>Picard (v.2.20.4)</td><td><a href="https://broadinstitute.github.io/picard/" target="_blank" rel="noopener noreferrer">Broad Institute</a></td></tr><tr><td>Transcript Quantification</td><td>featureCounts (utilities for counting reads to genomic features)</td><td><a href="http://subread.sourceforge.net/" target="_blank" rel="noopener noreferrer">featureCounts(v2.0.2)</a></td></tr><tr><td>Data Input File Format</td><td>File format in which sequencing data is provided</td><td><a href="https://academic.oup.com/nar/article/38/6/1767/3112533" target="_blank" rel="noopener noreferrer">FASTQ</a></td></tr><tr><td>Data Output File Formats</td><td>File formats in which Smart-seq2 output is provided</td><td><a href="http://samtools.github.io/hts-specs/" target="_blank" rel="noopener noreferrer">BAM</a>, Loom (counts and metrics; generated with <a href="http://loompy.org/" target="_blank" rel="noopener noreferrer">Loompy v.3.0.6)</a>, TSV (counts)</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="set-up"></a>Set-Up<a class="hash-link" href="#set-up" title="Direct link to heading">#</a></h2><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="smart-seq2-single-nucleus-installation-and-requirements"></a>Smart-seq2 Single Nucleus Installation and Requirements<a class="hash-link" href="#smart-seq2-single-nucleus-installation-and-requirements" title="Direct link to heading">#</a></h3><p>The snSS2 workflow code can be downloaded by cloning the GitHub <a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer">WARP repository</a>. For the latest release of snSS2, please see release tags prefixed with SmartSeq2SingleNucleus on the <a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer">WARP release page</a>.</p><p>The workflow is deployed using <a href="https://github.com/broadinstitute/cromwell" target="_blank" rel="noopener noreferrer">Cromwell</a>, a GA4GH compliant, flexible workflow management system that supports multiple computing platforms.</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Try the snSS2 workflow in Terra</h5></div><div class="admonition-content"><p>You can run the <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_nucleus_multisample/MultiSampleSmartSeq2SingleNucleus.wdl" target="_blank" rel="noopener noreferrer">Smart-seq2 Single Nucleus Multi-Sample workflow</a> (a wrapper for the snSS2 workflow described in this document) in <a href="https://app.terra.bio" target="_blank" rel="noopener noreferrer">Terra</a>, a cloud-based analysis platform. The Terra <a href="https://app.terra.bio/#workspaces/warp-pipelines/Smart-seq2_Single_Nucleus_Muti-Sample" target="_blank" rel="noopener noreferrer">Smart-seq2 Single Nucleus Multi-Sample public workspace</a> is preloaded with the Multi-snSS2 workflow, example testing data, and all the required reference data.</p></div></div><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="inputs"></a>Inputs<a class="hash-link" href="#inputs" title="Direct link to heading">#</a></h3><p>There is an <a href="https://github.com/broadinstitute/warp/tree/develop/pipelines/skylab/smartseq2_single_nucleus/mouse_paired_example.json" target="_blank" rel="noopener noreferrer">example configuration (JSON) file</a> available that you can to test the snSS2 workflow. It points to publicly available reference files and sample paired-end FASTQs.</p><h4><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="sample-data-input"></a>Sample Data Input<a class="hash-link" href="#sample-data-input" title="Direct link to heading">#</a></h4><p>The pipeline is designed for paired-end reads in the form of two FASTQ files. It processes one sample (cell) at a time.</p><ul><li><code>fastq1</code>: forward reads for a sample with paired-end sequencing </li><li><code>fastq2</code>: reverse reads for a sample with paired-end sequencing </li></ul><h4><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="reference-and-additional-inputs"></a>Reference and Additional Inputs<a class="hash-link" href="#reference-and-additional-inputs" title="Direct link to heading">#</a></h4><p>The snSS2 workflow requires multiple references detailed in the table below. </p><ul><li>Reference inputs are created using the <a href="https://github.com/broadinstitute/warp/tree/master/pipelines/skylab/build_indices" target="_blank" rel="noopener noreferrer">BuildIndices Pipeline</a>.</li><li>The workflow uses a modified version of the 10x Genomic&#x27;s code for building mouse (<a href="https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build#mm10_2020A" target="_blank" rel="noopener noreferrer">GRCm38-2020-A</a>) and human (<a href="https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build#GRCh38_2020A" target="_blank" rel="noopener noreferrer">GRCh38-2020-A</a>) reference packages. </li><li>To enable intron counting, the workflow calls a <a href="https://github.com/broadinstitute/warp/blob/develop/dockers/skylab/snss2-build-indices/add-introns-to-gtf.py" target="_blank" rel="noopener noreferrer">shell script</a> to create a custom GTF with intron annotations. Introns are considered any part of a contig that is not exonic nor intergenic. </li></ul><table><thead><tr><th>Reference Name</th><th>Reference Description</th><th>Format</th></tr></thead><tbody><tr><td>tar_star_reference</td><td>Genome references for STAR alignment</td><td>TAR</td></tr><tr><td>annotations_gtf</td><td>Custom GTF file containing annotations for exon and intron tagging; must match the STAR reference</td><td>GTF</td></tr><tr><td>genome_ref_fasta</td><td>FASTA file used for STAR alignment</td><td>FASTA</td></tr><tr><td>input_id</td><td>Unique identifier string for the cell; can be a UUID or human-readable ID</td><td>String</td></tr><tr><td>input_id_metadata_field</td><td>Optional value that can be used to identify any metadata fields specific to consortia (BRAIN Initiative, HCA, etc. ) that were used for the input_id</td><td>String</td></tr><tr><td>input_name</td><td>Optional unique identifier string for the cell; can be a human-readable alternative if a UUID is used for the input_id</td><td>String</td></tr><tr><td>input_name_metadata_field</td><td>Optional value that can be used to identify any metadata fields particular to consortia (e.g. Brain Initiative, HCA, etc. ) that were used for the input_name</td><td>String</td></tr><tr><td>output_name</td><td>Unique name used for output files; can also be a cloud path</td><td>String</td></tr><tr><td>adapter_list</td><td>File listing adapter sequences used in the library preparation (i.e. Illumina adapters for Illumina sequencing)</td><td>FASTA</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="running-smart-seq2"></a>Running Smart-seq2<a class="hash-link" href="#running-smart-seq2" title="Direct link to heading">#</a></h2><p>The <a href="https://github.com/broadinstitute/warp/tree/develop/pipelines/skylab/smartseq2_single_nucleus/SmartSeq2SingleNucleus.wdl" target="_blank" rel="noopener noreferrer">snSS2 workflow </a> is in the <a href="https://github.com/broadinstitute/warp/tree/snSS2_first_wdls/pipelines/skylab/smartseq2_single_nucleus" target="_blank" rel="noopener noreferrer">pipelines/smartseq2_single_nucleus folder</a> of the WARP repository and implements the workflow by importing individual tasks (written in WDL script) from the WARP <a href="https://github.com/broadinstitute/warp/tree/master/tasks/skylab/" target="_blank" rel="noopener noreferrer">tasks folder</a>.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="smart-seq2-single-nucleus-workflow-summary"></a>Smart-seq2 Single Nucleus Workflow Summary<a class="hash-link" href="#smart-seq2-single-nucleus-workflow-summary" title="Direct link to heading">#</a></h3><table><thead><tr><th>Task Name and Task’s WDL Link</th><th>Description</th><th>Software</th><th>Tool</th></tr></thead><tbody><tr><td><a href="https://github.com/broadinstitute/warp/tree/develop/tasks/skylab/TrimAdapters.wdl" target="_blank" rel="noopener noreferrer">TrimAdapters.TrimAdapters</a></td><td>Trims adapter sequences from the FASTQ inputs</td><td><a href="https://github.com/ExpressionAnalysis/ea-utils" target="_blank" rel="noopener noreferrer">ea-utils</a></td><td><a href="https://github.com/ExpressionAnalysis/ea-utils/tree/master/clipper" target="_blank" rel="noopener noreferrer">fastq-mcf</a></td></tr><tr><td><a href="https://github.com/broadinstitute/warp/tree/develop/tasks/skylab/StarAlign.wdl" target="_blank" rel="noopener noreferrer">StarAlignFastq.StarAlignFastqPairedEnd</a></td><td>Aligns reads to the genome</td><td>STAR</td><td>STAR</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/tree/develop/tasks/skylab/Picard.wdl" target="_blank" rel="noopener noreferrer">Picard.RemoveDuplicatesFromBam</a></td><td>Removes duplicate reads, producing a new BAM output; adds regroups to deduplicated BAM</td><td>Picard</td><td>MarkDuplicates, AddOrReplaceReadGroups</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/tree/develop/tasks/skylab/Picard.wdl" target="_blank" rel="noopener noreferrer">Picard.CollectMultipleMetrics</a></td><td>Collects QC metrics on the deduplicated BAM file</td><td>Picard</td><td>CollectMultipleMetrics</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/tree/develop/tasks/skylab/FeatureCounts.wdl" target="_blank" rel="noopener noreferrer">CountAlignments.CountAlignments</a></td><td>Uses a custom GTF with featureCounts and Python to mark introns, create a BAM that has alignments spanning intron-exon junctions removed, and counts exons using the custom BAM and by excluding intron tags</td><td><a href="http://subread.sourceforge.net/" target="_blank" rel="noopener noreferrer">Subread</a></td><td><a href="http://bioinf.wehi.edu.au/featureCounts/" target="_blank" rel="noopener noreferrer">FeatureCounts</a>, Python 3</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/tree/develop/tasks/skylab/LoomUtils.wdl" target="_blank" rel="noopener noreferrer">LoomUtils.SingleNucleiSmartSeq2LoomOutput</a></td><td>Combines featureCounts and Picard metric outputs into a Loom formatted cell-by-gene matrix</td><td>Python 3</td><td><a href="https://github.com/broadinstitute/warp/tree/develop/dockers/skylab/loom-output/create_loom_snss2.py" target="_blank" rel="noopener noreferrer">Custom script</a></td></tr></tbody></table><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="trimming-adapters"></a>Trimming Adapters<a class="hash-link" href="#trimming-adapters" title="Direct link to heading">#</a></h3><p>The TrimAdapters task uses the adapter list reference file to run the <a href="https://github.com/ExpressionAnalysis/ea-utils/tree/master/clipper" target="_blank" rel="noopener noreferrer">fastq-mcf</a> tool. This tool identifies the adapters in the input FASTQ files and performs clipping by using a subsampling parameter of 200,000 reads. The task outputs the trimmed FASTQ files which are then used for alignment. </p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="aligning-reads"></a>Aligning Reads<a class="hash-link" href="#aligning-reads" title="Direct link to heading">#</a></h3><p>The StarAlignFastq task runs the STAR aligner on the trimmed FASTQ files. The STAR quantMode parameter is set to GeneCounts, which counts the number of reads per gene while mapping. The task outputs a coordinate-sorted aligned BAM file. </p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="removing-duplicate-reads"></a>Removing Duplicate Reads<a class="hash-link" href="#removing-duplicate-reads" title="Direct link to heading">#</a></h3><p>The RemoveDuplicatesFromBam task removes multi-mapped reads, optical duplicates, and PCR duplicates from the aligned BAM. It then adds readgroup information and creates a new, coordinate-sorted aligned BAM output.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="collecting-metrics"></a>Collecting Metrics<a class="hash-link" href="#collecting-metrics" title="Direct link to heading">#</a></h3><p>The CollectMultipleMetrics task uses the Picard tool CollectMultipleMetrics to perform QC on the deduplicated BAM file. These metrics are copied to the final cell-by-gene matrix output (Loom file).</p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="counting-genes"></a>Counting Genes<a class="hash-link" href="#counting-genes" title="Direct link to heading">#</a></h3><p>The CountAlignments task uses the featureCounts package to count introns and exons. First, the featureCounts tool counts intronic alignments in the deduplicated BAM using a custom GTF with annotated introns. The tool flags intronic alignments if they overlap an annotated intron by a minimum of 3 bp. </p><p>Next, following pipeline processes established by the BICCN Whole Mouse Brain Working Group, a custom Python script (“<a href="https://github.com/broadinstitute/warp/tree/develop/dockers/skylab/featureCounts/remove-reads-on-junctions.py" target="_blank" rel="noopener noreferrer">remove-reads-on-junctions.py</a>”) removes all alignments in the deduplicated BAM that cross only one intron-exon junction and produces an intermediate BAM file for exon counting. This removes a negligible amount of putative intronic alignments that did not meet the 3 bp intron overlap criteria. </p><p>Lastly, featureCounts uses the intermediate BAM with junctions removed to count exons. The final outputs of this step include a cell-by-gene matrix of intronic counts, a cell-by-gene matrix of exonic counts, and summary metrics for the different count types.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="creating-the-cell-by-gene-matrix-loom"></a>Creating the Cell-by-Gene Matrix (Loom)<a class="hash-link" href="#creating-the-cell-by-gene-matrix-loom" title="Direct link to heading">#</a></h3><p>The LoomUtils task combines the Picard metrics (alignment_summary_metrics, deduplication metrics, and the G/C bias summary metrics) with the featureCount exon and intron counts to create a Loom formatted cell-by-gene count matrix. </p><p>Exonic counts are in the Loom matrix and intronic counts are added as a Loom layer. Read more about Loom file format in the <a href="https://linnarssonlab.org/loompy/index.html" target="_blank" rel="noopener noreferrer">Loompy documentation</a>.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="outputs"></a>Outputs<a class="hash-link" href="#outputs" title="Direct link to heading">#</a></h3><p>The table below details the final outputs of the snSS2 workflow.  </p><table><thead><tr><th>Output Name</th><th>Output Description</th><th>Output Format</th></tr></thead><tbody><tr><td>pipeline_version_out</td><td>Version of the processing pipeline run on this data</td><td>String</td></tr><tr><td>aligned_bam</td><td>Aligned BAM</td><td>BAM</td></tr><tr><td>exon_intron_counts</td><td>Text file containing intron counts, intron FPKMs, exon counts, and exon FPKMs</td><td>TXT</td></tr><tr><td>loom_output_files</td><td>Cell-by-gene count matrix that includes the raw exon counts (in matrix), intron counts (in matrix layer), cell metrics (column attributes) and gene IDs (row attributes)</td><td>Loom</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="validation"></a>Validation<a class="hash-link" href="#validation" title="Direct link to heading">#</a></h2><p>The snSS2 pipeline was scientifically validated by the BRAIN Initiatives Cell Census Network (BICCN) 2.0 Whole Mouse Brain Working Group. </p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="versioning"></a>Versioning<a class="hash-link" href="#versioning" title="Direct link to heading">#</a></h2><p>All Smart-seq2 Single Nucleus release notes are documented in the <a href="https://github.com/broadinstitute/warp/tree/develop/pipelines/skylab/smartseq2_single_nucleus/SmartSeq2SingleNucleus.changelog.md" target="_blank" rel="noopener noreferrer">Smartseq2 Single Nucleus changelog</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="consortia-support"></a>Consortia Support<a class="hash-link" href="#consortia-support" title="Direct link to heading">#</a></h2><p>This pipeline is supported and used by the <a href="https://biccn.org/" target="_blank" rel="noopener noreferrer">BRAIN Initiative Cell Census Network</a> (BICCN). </p><p>If your organization also uses this pipeline, we would love to list you! Please reach out to us by contacting <a href="mailto:kdegatano@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kylee Degatano</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="have-suggestions"></a>Have Suggestions?<a class="hash-link" href="#have-suggestions" title="Direct link to heading">#</a></h2><p>Help us make our tools better by contacting <a href="mailto:kdegatano@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kylee Degatano</a> for pipeline-related suggestions or questions.</p></div><footer class="row docusaurus-mt-lg"><div class="col"><a href="https://github.com/broadinstitute/warp/edit/develop/website/docs/Pipelines/Smart-seq2_Single_Nucleus_Pipeline/README.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_2_ui" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_3DPF">Last updated on <b><time datetime="2021-07-06T13:49:23.000Z">7/6/2021</time></b> by <b>ekiernan</b></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/README"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">« Smart-seq2 Single Nucleus Multi-Sample Overview</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/README"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Smart-seq2 Single Sample Overview »</div></a></div></nav></div></div><div class="col col--3"><div class="tableOfContents_35-E thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#introduction-to-the-smart-seq2-single-nucleus-pipeline" class="table-of-contents__link">Introduction to the Smart-seq2 Single Nucleus Pipeline</a></li><li><a href="#quick-start-table" class="table-of-contents__link">Quick Start Table</a></li><li><a href="#set-up" class="table-of-contents__link">Set-Up</a><ul><li><a href="#smart-seq2-single-nucleus-installation-and-requirements" class="table-of-contents__link">Smart-seq2 Single Nucleus Installation and Requirements</a></li><li><a href="#inputs" class="table-of-contents__link">Inputs</a></li></ul></li><li><a href="#running-smart-seq2" class="table-of-contents__link">Running Smart-seq2</a><ul><li><a href="#smart-seq2-single-nucleus-workflow-summary" class="table-of-contents__link">Smart-seq2 Single Nucleus Workflow Summary</a></li><li><a href="#trimming-adapters" class="table-of-contents__link">Trimming Adapters</a></li><li><a href="#aligning-reads" class="table-of-contents__link">Aligning Reads</a></li><li><a href="#removing-duplicate-reads" class="table-of-contents__link">Removing Duplicate Reads</a></li><li><a href="#collecting-metrics" class="table-of-contents__link">Collecting Metrics</a></li><li><a href="#counting-genes" class="table-of-contents__link">Counting Genes</a></li><li><a href="#creating-the-cell-by-gene-matrix-loom" class="table-of-contents__link">Creating the Cell-by-Gene Matrix (Loom)</a></li><li><a href="#outputs" class="table-of-contents__link">Outputs</a></li></ul></li><li><a href="#validation" class="table-of-contents__link">Validation</a></li><li><a href="#versioning" class="table-of-contents__link">Versioning</a></li><li><a href="#consortia-support" class="table-of-contents__link">Consortia Support</a></li><li><a href="#have-suggestions" class="table-of-contents__link">Have Suggestions?</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/get-started">Get Started</a></li></ul></div><div class="col footer__col"><div class="footer__title">Guide and Policy</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/contribution/README">Contribution Guide</a></li><li class="footer__item"><a class="footer__link-item" href="/warp/privacy">Privacy</a></li></ul></div><div class="col footer__col"><div class="footer__title">Resources</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/blog">Blog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer" class="footer__link-item">Changelog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub</a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2021 Copyright © Data Sciences Platform, Broad Institute.</div></div></div></footer></div>
<script src="/warp/assets/js/runtime~main.6a48d3f3.js"></script>
<script src="/warp/assets/js/main.41bb09f5.js"></script>
</body>
</html>