<!doctype html>
<html lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.0">
<link rel="alternate" type="application/rss+xml" href="/warp/blog/rss.xml" title="WARP Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/warp/blog/atom.xml" title="WARP Blog Atom Feed">
<link rel="preconnect" href="https://www.google-analytics.com">
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-181048107-1","auto"),ga("set","anonymizeIp",!0),ga("send","pageview")</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
<script>!function(t,h,e,j,s,n){t.hj=t.hj||function(){(t.hj.q=t.hj.q||[]).push(arguments)},t._hjSettings={hjid:2427684,hjsv:6},s=h.getElementsByTagName("head")[0],(n=h.createElement("script")).async=1,n.src="https://static.hotjar.com/c/hotjar-"+t._hjSettings.hjid+".js?sv="+t._hjSettings.hjsv,s.appendChild(n)}(window,document)</script><title data-react-helmet="true">Smart-seq2 Multi-Sample Overview | WARP</title><meta data-react-helmet="true" property="og:url" content="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/README"><meta data-react-helmet="true" name="docusaurus_locale" content="en"><meta data-react-helmet="true" name="docusaurus_version" content="current"><meta data-react-helmet="true" name="docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Smart-seq2 Multi-Sample Overview | WARP"><meta data-react-helmet="true" name="description" content="| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |"><meta data-react-helmet="true" property="og:description" content="| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |"><link data-react-helmet="true" rel="shortcut icon" href="/warp/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/README"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/README" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/README" hreflang="x-default"><link rel="stylesheet" href="/warp/assets/css/styles.6c5d82f2.css">
<link rel="preload" href="/warp/assets/js/runtime~main.595f6845.js" as="script">
<link rel="preload" href="/warp/assets/js/main.4039a7a9.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#main" class="skipToContent_1oUP">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle" type="button" tabindex="0"><svg aria-label="Menu" width="30" height="30" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/warp/"><strong class="navbar__title">WARP Pipelines</strong></a></div><div class="navbar__items navbar__items--right"><a class="navbar__item navbar__link navbar__link--active" href="/warp/docs/get-started">Documentation</a><a class="navbar__item navbar__link" href="/warp/blog">Blog</a><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link" aria-label="GitHub repository"></a><div class="react-toggle displayOnlyInLargeViewport_GrZ2 react-toggle--disabled" role="button" tabindex="-1"><div class="react-toggle-track"><div class="react-toggle-track-check"><span class="toggle_71bT">ðŸŒœ</span></div><div class="react-toggle-track-x"><span class="toggle_71bT">ðŸŒž</span></div></div><div class="react-toggle-thumb"></div><input type="checkbox" class="react-toggle-screenreader-only" aria-label="Switch between dark and light mode"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div><div class="navbar-sidebar"><div class="navbar-sidebar__brand"><a class="navbar__brand" href="/warp/"><strong class="navbar__title">WARP Pipelines</strong></a></div><div class="navbar-sidebar__items"><div class="menu"><ul class="menu__list"><li class="menu__list-item"><a class="menu__link navbar__link--active" href="/warp/docs/get-started">Documentation</a></li><li class="menu__list-item"><a class="menu__link" href="/warp/blog">Blog</a></li><li class="menu__list-item"><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="menu__link header-github-link" aria-label="GitHub repository"></a></li></ul></div></div></div></nav><div class="main-wrapper docs-wrapper doc-page"><div class="docPage_31aa"><div class="docSidebarContainer_3Kbt" role="complementary"><div class="sidebar_15mo"><div class="menu menu--responsive thin-scrollbar menu_Bmed"><button aria-label="Open menu" aria-haspopup="true" class="button button--secondary button--sm menu__button" type="button"><svg aria-label="Menu" class="sidebarMenuIcon_fgN0" width="24" height="24" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/warp/docs/get-started">Welcome to WARP</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">About WARP</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/About_WARP/BestPractices">Best Practices for Building Data Processing Pipelines</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/About_WARP/PipelineRequirements">Pipeline Requirements</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/About_WARP/TestingPipelines">Testing Pipelines</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/About_WARP/VersionAndReleasePipelines">Version and Release Pipelines</a></li></ul></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#!">WARP Pipelines</a><ul class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">CEMBA</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/CEMBA_MethylC_Seq_Pipeline/README">CEMBA Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/CEMBA_MethylC_Seq_Pipeline/CEMBA.methods">CEMBA_v1.1.0 Publication Methods</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">Exome Germline Single Sample</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Exome_Germline_Single_Sample_Pipeline/README">Exome Germline Single Sample Overview</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">GDC Whole Genome Somatic Single Sample</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Genomic_Data_Commons_Whole_Genome_Somatic/README">Genomic Data Commons (GDC) Whole Genome Somatic Single Sample Overview</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">Illumina Genotyping Array</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Illumina_Genotyping_Arrays_Pipeline/IlluminaGenotypingArray.documentation">Illumina Genotyping Array Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Illumina_Genotyping_Arrays_Pipeline/Illumina_genotyping_array_spec">VCF Overview: Illumina Genotyping Array</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">Optimus</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Optimus_Pipeline/README">Optimus Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Optimus_Pipeline/optimus.methods">Optimus v4.2.4 Methods</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Optimus_Pipeline/Bam_tags">What tags are included in an Optimus BAM file?</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Optimus_Pipeline/Loom_schema">Optimus Count Matrix Overview</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">Single Cell ATAC</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Single_Cell_ATAC_Seq_Pipeline/README">scATAC Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Single_Cell_ATAC_Seq_Pipeline/scatac.methods">scATAC v1.2.0 Methods</a></li></ul></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#!" tabindex="0">Smart-seq2 Multi-Sample</a><ul class="menu__list"><li class="menu__list-item"><a aria-current="page" class="menu__link menu__link--active active" tabindex="0" href="/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/README">Smart-seq2 Multi-Sample Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/smart-seq2.methods">Smart-seq2 Multi-Sample v2.2.1 Publication Methods</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/Loom_schema">What&#x27;s in the Smart-seq2 Multi Sample Pipeline Loom File?</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">Smart-seq2 Single Nucleus Multi-Sample</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/README">Smart-seq2 Single Nucleus Multi-Sample Overview</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">Smart-seq2 Single Nucleus</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Pipeline/README">Smart-seq2 Single Nucleus Overview</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">Smart-seq2 Single Sample</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/README">Smart-seq2 Single Sample Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/Loom_schema">What&#x27;s in the Smart-seq2 Single Sample Pipeline Loom File?</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="0">Whole Genome Germline Single Sample</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Whole_Genome_Germline_Single_Sample_Pipeline/README">Whole Genome Germline Single Sample Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/Pipelines/Whole_Genome_Germline_Single_Sample_Pipeline/wgs.methods">Whole Genome Germline Single Sample v2.2.0 Methods</a></li></ul></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Contribution Guide</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/contribution/README">Welcome</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="-1">Contribute to WARP</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/contribution/contribute_to_warp/changelog_style">Changelog Style Guide Overview</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!" tabindex="-1">Contribute to WARP Documentation</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/contribution/contribute_to_warp_docs/docsite_maintenance">WARP Documentation Maintenance Guide</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/warp/docs/contribution/contribute_to_warp_docs/doc_style">Documentation Style Guide</a></li></ul></li></ul></li></ul></div></div></div><main class="docMainContainer_3ufF"><div class="container padding-vert--lg docItemWrapper_3FMP"><div class="row"><div class="col docItemCol_3FnS"><div class="docItemContainer_33ec"><article><header><h1 class="docTitle_3a4h">Smart-seq2 Multi-Sample Overview</h1></header><div class="markdown"><table><thead><tr><th align="center">Pipeline Version</th><th align="center">Date Updated</th><th align="center">Documentation Author</th><th align="center">Questions or Feedback</th></tr></thead><tbody><tr><td align="center"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer">MultiSampleSmartSeq2_v2.2.1</a></td><td align="center">May, 2021</td><td align="center"><a href="mailto:ekiernan@broadinstitute.org" target="_blank" rel="noopener noreferrer">Elizabeth Kiernan</a></td><td align="center">Please file GitHub issues in WARP or contact <a href="mailto:kdegatano@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kylee Degatano</a></td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="introduction"></a>Introduction<a class="hash-link" href="#introduction" title="Direct link to heading">#</a></h2><p>The Smart-seq2 Multi-Sample (Multi-SS2) Pipeline is a wrapper around the <a href="/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/README">Smart-seq2 Single Sample</a> pipeline. It is developed by the Data Coordination Platform of the Human Cell Atlas to process single-cell RNAseq (scRNAseq) data generated by Smart-seq2 assays. The workflow processes multiple cells by importing and running the <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.wdl" target="_blank" rel="noopener noreferrer">Smart-seq2 Single Sample workflow</a> for each cell (sample) and then merging the resulting Loom matrix output into a single Loom matrix containing raw counts and TPMs.</p><p>Full details about the Smart-seq2 Pipeline can be read in the <a href="/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/README">Smart-seq2 Single Sample Overview</a> in GitHub.</p><p>The Multi-SS2 workflow can also be run in <a href="https://app.terra.bio" target="_blank" rel="noopener noreferrer">Terra</a>, a cloud-based analysis platform. The Terra <a href="https://app.terra.bio/#workspaces/featured-workspaces-hca/HCA%20Smart-seq2%20Multi%20Sample%20Pipeline" target="_blank" rel="noopener noreferrer">Smart-seq2 public workspace</a> contains the Smart-seq2 workflow, workflow configurations, required reference data and other inputs, and example testing data.</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Want to use the Multi-SS2 Pipeline for your publication?</h5></div><div class="admonition-content"><p>Check out the <a href="/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/smart-seq2.methods">Smart-seq2 Publication Methods</a> to get started!</p></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="inputs"></a>Inputs<a class="hash-link" href="#inputs" title="Direct link to heading">#</a></h2><p>There are two example configuration (JSON) files available for testing the Multi-SS2 workflow. Both examples are also preloaded in the Terra <a href="https://app.terra.bio/#workspaces/featured-workspaces-hca/HCA%20Smart-seq2%20Multi%20Sample%20Pipeline" target="_blank" rel="noopener noreferrer">Smart-seq2 public workspace</a>.</p><ul><li><a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_multisample/human_single_example.json" target="_blank" rel="noopener noreferrer">human_single_example.json</a>: Configurations for an example single-end human dataset consisting of two samples (cells)</li><li><a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_multisample/mouse_paired_example.json" target="_blank" rel="noopener noreferrer">mouse_paired_example.json</a>: Configurations for an example paired-end mouse dataset consisting of two samples (cells)</li></ul><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="sample-and-reference-inputs"></a>Sample and Reference Inputs<a class="hash-link" href="#sample-and-reference-inputs" title="Direct link to heading">#</a></h3><p>The workflowâ€™s sample inputs are listed in the table below. Reference inputs are identical to those
specified in the <a href="/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/README">Smart-seq2 Single Sample Overview</a>.</p><p>The workflow processes both single- and paired-end samples; however, these samples can not be mixed in the same run.</p><table><thead><tr><th>Input name</th><th>Input Description</th><th>Input Type</th></tr></thead><tbody><tr><td>fastq1_input_files</td><td>Cloud locations for each read1 file</td><td>Array of strings</td></tr><tr><td>fastq2_input_files</td><td>Optional cloud locations for each read2 file if running paired-end samples</td><td>Array of strings</td></tr><tr><td>input_ids</td><td>Unique identifiers or names for each cell; can be a UUID or human-readable name</td><td>Array of strings</td></tr><tr><td>input_names</td><td>Optional unique identifiers/names to further describe each cell. If <code>input_id</code> is a UUID, the <code>input_name</code> could be used as a human-readable identifier</td><td>String</td></tr><tr><td>batch_id</td><td>Identifier for the batch of multiple samples</td><td>String</td></tr><tr><td>batch_name</td><td>Optional string to describe the batch or biological sample</td><td>String</td></tr><tr><td>input_name_metadata_field</td><td>Optional input describing, when applicable, the metadata field containing the <code>input_names</code></td><td>String</td></tr><tr><td>input_id_metadata_field</td><td>Optional string describing, when applicable, the metadata field containing the <code>input_ids</code></td><td>String</td></tr><tr><td><code>project_id</code></td><td>Optional project identifier; usually a number</td><td>String</td></tr><tr><td><code>project_name</code></td><td>Optional project identifier; usually a human-readable name</td><td>String</td></tr><tr><td><code>library</code></td><td>Optional description of the sequencing method or approach</td><td>String</td></tr><tr><td><code>organ</code></td><td>Optional description of the organ from which the cells were derived</td><td>String</td></tr><tr><td><code>species</code></td><td>Optional description of the species from which the cells were derived</td><td>String</td></tr><tr><td><code>paired-end</code></td><td>Boolean for whether samples are paired-end or not</td><td>Boolean</td></tr></tbody></table><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="additional-input"></a>Additional Input<a class="hash-link" href="#additional-input" title="Direct link to heading">#</a></h3><p>The reference inputs are identical to those specified in the &quot;Additional Reference Inputs&quot; section of the <a href="https://broadinstitute.github.io/warp/documentation/Pipelines/Smart-seq2_Single_Sample_Pipeline/#inputs" target="_blank" rel="noopener noreferrer">Smart-seq2 Single Sample Overview</a>.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="smart-seq2-multi-sample-task-summary"></a>Smart-seq2 Multi-Sample Task Summary<a class="hash-link" href="#smart-seq2-multi-sample-task-summary" title="Direct link to heading">#</a></h3><p>The Multi-SS2 Pipeline calls three tasks:</p><p>1) <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.wdl" target="_blank" rel="noopener noreferrer">SmartSeq2SingleSample.wdl</a>: a task that runs the Smart-seq2 Single Sample workflow
2) <a href="https://github.com/broadinstitute/warp/blob/master/tasks/SmartSeq2PlateAggregation.wdl" target="_blank" rel="noopener noreferrer">SmartSeq2PlateAggregation.wdl</a>: the wrapper pipeline that checks the inputs, initiates per cell processing and aggregates the results
3) <a href="https://github.com/broadinstitute/warp/blob/master/tasks/LoomUtils.wdl" target="_blank" rel="noopener noreferrer">LoomUtils.wdl</a>: a task that produces the Loom matrix output containing raw counts and TPMs</p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="outputs"></a>Outputs<a class="hash-link" href="#outputs" title="Direct link to heading">#</a></h3><table><thead><tr><th>Output file name</th><th>Output Description</th><th>Output Type</th></tr></thead><tbody><tr><td>bam_files</td><td>An array of genome-aligned BAM files (one for each sample) generated with HISAT2</td><td>Array</td></tr><tr><td>bam_index_files</td><td>An array of BAM index files generated with HISAT2</td><td>Array</td></tr><tr><td>loom_output</td><td>A single Loom cell-by-gene matrix containing raw counts and TPMs for every cell</td><td>File</td></tr></tbody></table><p>The final Loom matrix is an aggregate of all the individual Loom matrices generated using the <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.wdl" target="_blank" rel="noopener noreferrer">Smart-seq2 Single Sample workflow</a>.</p><p>The aggregated Loom filename contains the <code>batch_id</code> prefix, which is the string specified in the input configuration. The <code>batch_id</code> is also set as a global attribute in the Loom.</p><p>Both the individual sample Loom files and individual BAM files are described in the <a href="/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/README">Smart-seq2 Single Sample Overview</a>.</p><div class="admonition admonition-warning alert alert--danger"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M5.05.31c.81 2.17.41 3.38-.52 4.31C3.55 5.67 1.98 6.45.9 7.98c-1.45 2.05-1.7 6.53 3.53 7.7-2.2-1.16-2.67-4.52-.3-6.61-.61 2.03.53 3.33 1.94 2.86 1.39-.47 2.3.53 2.27 1.67-.02.78-.31 1.44-1.13 1.81 3.42-.59 4.78-3.42 4.78-5.56 0-2.84-2.53-3.22-1.25-5.61-1.52.13-2.03 1.13-1.89 2.75.09 1.08-1.02 1.8-1.86 1.33-.67-.41-.66-1.19-.06-1.78C8.18 5.31 8.68 2.45 5.05.32L5.03.3l.02.01z"></path></svg></span>Zarr Array Deprecation Notice June 2020</h5></div><div class="admonition-content"><p>Please note that we have deprecated the previously used Zarr array output. The pipeline now uses the Loom file format as the default output.</p></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="validation"></a>Validation<a class="hash-link" href="#validation" title="Direct link to heading">#</a></h2><p>The Multi-SS2 Pipeline has been validated for processing human and mouse, stranded or unstranded, paired- or single-end, and plate- or fluidigm-based Smart-seq2 datasets (see links to validation reports in the table below).</p><table><thead><tr><th>Workflow Configuration</th><th>Link to Report</th></tr></thead><tbody><tr><td>Mouse paired-end</td><td><a href="https://docs.google.com/document/d/12zGTFROrcXEByt9z0h06qjSqb9vWutn28Tx6YiND1Ds/edit" target="_blank" rel="noopener noreferrer">Report</a></td></tr><tr><td>Human and mouse single-end</td><td><a href="https://docs.google.com/document/d/1MonsTG8UnROHZ_XpulrSZNTxO988KEH6T6h45plFYQg/edit#heading=h.ixoqmhbabdvh" target="_blank" rel="noopener noreferrer">Report</a></td></tr><tr><td>Human stranded fluidigm</td><td><a href="https://docs.google.com/document/d/1FEg86Tlu657j9Kjw_v3keFQRXcBIs8gOqCwLbPSP-C0/edit#heading=h.wjr8otl7zg14" target="_blank" rel="noopener noreferrer">Report</a></td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="versioning"></a>Versioning<a class="hash-link" href="#versioning" title="Direct link to heading">#</a></h2><p>Release information for the Multi-SS2 Pipeline can be found in the <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_multisample/MultiSampleSmartSeq2.changelog.md" target="_blank" rel="noopener noreferrer">changelog</a>. Please note that any major changes to the Smart-seq2 pipeline will be documented in the <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.changelog.md" target="_blank" rel="noopener noreferrer">Smart-seq2 Single Sample changelog</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="citing-the-smart-seq2-multi-sample-pipeline"></a>Citing the Smart-seq2 Multi-Sample Pipeline<a class="hash-link" href="#citing-the-smart-seq2-multi-sample-pipeline" title="Direct link to heading">#</a></h2><p>Please identify the pipeline in your methods section using the Smart-seq2 Multi-Sample Pipeline&#x27;s <a href="https://scicrunch.org/scicrunch/Resources/record/nlx_144509-1/SCR_018920/resolver?q=Smart-seq2&amp;l=Smart-seq2" target="_blank" rel="noopener noreferrer">SciCrunch resource identifier</a>.</p><ul><li>Ex: <em>Smart-seq2 Multi-Sample Pipeline (RRID:SCR_018920)</em></li></ul><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="have-suggestions"></a>Have Suggestions?<a class="hash-link" href="#have-suggestions" title="Direct link to heading">#</a></h2><p>Please help us make our tools better by contacting <a href="mailto:kdegatano@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kylee Degatano</a> for pipeline-related suggestions or questions.</p></div></article><div class="margin-vert--xl"><div class="row"><div class="col"><a href="https://github.com/facebook/docusaurus/edit/master/website/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/README.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="1.2em" width="1.2em" preserveAspectRatio="xMidYMid meet" role="img" viewBox="0 0 40 40" class="iconEdit_2_ui" aria-label="Edit page"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div></div></div><div class="margin-vert--lg"><nav class="pagination-nav" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Single_Cell_ATAC_Seq_Pipeline/scatac.methods"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Â« scATAC v1.2.0 Methods</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/smart-seq2.methods"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Smart-seq2 Multi-Sample v2.2.1 Publication Methods Â»</div></a></div></nav></div></div></div><div class="col col--3"><div class="tableOfContents_35-E thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#introduction" class="table-of-contents__link">Introduction</a></li><li><a href="#inputs" class="table-of-contents__link">Inputs</a><ul><li><a href="#sample-and-reference-inputs" class="table-of-contents__link">Sample and Reference Inputs</a></li><li><a href="#additional-input" class="table-of-contents__link">Additional Input</a></li><li><a href="#smart-seq2-multi-sample-task-summary" class="table-of-contents__link">Smart-seq2 Multi-Sample Task Summary</a></li><li><a href="#outputs" class="table-of-contents__link">Outputs</a></li></ul></li><li><a href="#validation" class="table-of-contents__link">Validation</a></li><li><a href="#versioning" class="table-of-contents__link">Versioning</a></li><li><a href="#citing-the-smart-seq2-multi-sample-pipeline" class="table-of-contents__link">Citing the Smart-seq2 Multi-Sample Pipeline</a></li><li><a href="#have-suggestions" class="table-of-contents__link">Have Suggestions?</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><h4 class="footer__title">Docs</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/get-started">Get Started</a></li></ul></div><div class="col footer__col"><h4 class="footer__title">Guide and Policy</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/contribution/README">Contribution Guide</a></li><li class="footer__item"><a class="footer__link-item" href="/warp/privacy">Privacy</a></li></ul></div><div class="col footer__col"><h4 class="footer__title">Resources</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/blog">Blog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer" class="footer__link-item">Changelog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub</a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright Â© 2021 Copyright Â© Data Sciences Platform, Broad Institute.</div></div></div></footer></div>
<script src="/warp/assets/js/runtime~main.595f6845.js"></script>
<script src="/warp/assets/js/main.4039a7a9.js"></script>
</body>
</html>