<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Whole Genome Germline Single Sample v2.2.0 Methods | WARP</title>
    <meta name="generator" content="VuePress 1.7.1">
    <link rel="icon" href="/warp/logo.png">
    <meta name="description" content="WDL Analysis Research Pipelines">
    <meta name="theme-color" content="#3eaf7c">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <meta name="og:locale" content="en_US">
    
    <link rel="preload" href="/warp/assets/css/0.styles.647b5de0.css" as="style"><link rel="preload" href="/warp/assets/js/app.0913a268.js" as="script"><link rel="preload" href="/warp/assets/js/2.2043750e.js" as="script"><link rel="preload" href="/warp/assets/js/36.fcca5ed7.js" as="script"><link rel="prefetch" href="/warp/assets/js/10.a9482500.js"><link rel="prefetch" href="/warp/assets/js/11.e5f035f9.js"><link rel="prefetch" href="/warp/assets/js/12.d33b4dbc.js"><link rel="prefetch" href="/warp/assets/js/13.2d65679b.js"><link rel="prefetch" href="/warp/assets/js/14.44af373e.js"><link rel="prefetch" href="/warp/assets/js/15.8ef1d142.js"><link rel="prefetch" href="/warp/assets/js/16.bae06c91.js"><link rel="prefetch" href="/warp/assets/js/17.ed42aaa3.js"><link rel="prefetch" href="/warp/assets/js/18.3ed01c57.js"><link rel="prefetch" href="/warp/assets/js/19.a80d5c3b.js"><link rel="prefetch" href="/warp/assets/js/20.3ea0888d.js"><link rel="prefetch" href="/warp/assets/js/21.f2d33845.js"><link rel="prefetch" href="/warp/assets/js/22.e28d3bc6.js"><link rel="prefetch" href="/warp/assets/js/23.58fc4838.js"><link rel="prefetch" href="/warp/assets/js/24.1e044fd9.js"><link rel="prefetch" href="/warp/assets/js/25.74089735.js"><link rel="prefetch" href="/warp/assets/js/26.d9c25083.js"><link rel="prefetch" href="/warp/assets/js/27.ef3ce64a.js"><link rel="prefetch" href="/warp/assets/js/28.3b9f6f95.js"><link rel="prefetch" href="/warp/assets/js/29.a98e017d.js"><link rel="prefetch" href="/warp/assets/js/3.dee46bae.js"><link rel="prefetch" href="/warp/assets/js/30.31faf728.js"><link rel="prefetch" href="/warp/assets/js/31.9eed8869.js"><link rel="prefetch" href="/warp/assets/js/32.5eb9b971.js"><link rel="prefetch" href="/warp/assets/js/33.fc88493a.js"><link rel="prefetch" href="/warp/assets/js/34.a61f9f57.js"><link rel="prefetch" href="/warp/assets/js/35.abd40f98.js"><link rel="prefetch" href="/warp/assets/js/37.3fdfb55c.js"><link rel="prefetch" href="/warp/assets/js/38.d42f90f3.js"><link rel="prefetch" href="/warp/assets/js/39.709ece23.js"><link rel="prefetch" href="/warp/assets/js/4.e5554209.js"><link rel="prefetch" href="/warp/assets/js/5.961aca45.js"><link rel="prefetch" href="/warp/assets/js/6.57da0acf.js"><link rel="prefetch" href="/warp/assets/js/7.e72f1202.js"><link rel="prefetch" href="/warp/assets/js/8.1821baf8.js"><link rel="prefetch" href="/warp/assets/js/9.961e8595.js">
    <link rel="stylesheet" href="/warp/assets/css/0.styles.647b5de0.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/warp/" class="home-link router-link-active"><!----> <span class="site-name">WARP</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="/warp/documentation/Intro.html" class="nav-link">
  Documentation
</a></div><div class="nav-item"><a href="/warp/blog/" class="nav-link">
  Blog
</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="Learn More" class="dropdown-title"><span class="title">Learn More</span> <span class="arrow down"></span></button> <button type="button" aria-label="Learn More" class="mobile-dropdown-title"><span class="title">Learn More</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><h4>
          Explore
        </h4> <ul class="dropdown-subitem-wrapper"><li class="dropdown-subitem"><a href="/warp/research/" class="nav-link">
  Research
</a></li></ul></li><li class="dropdown-item"><h4>
          Resources
        </h4> <ul class="dropdown-subitem-wrapper"><li class="dropdown-subitem"><a href="/warp/contribution/" class="nav-link">
  Contribution Guide
</a></li><li class="dropdown-subitem"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer" class="nav-link external">
  Changelog
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></li><li class="dropdown-subitem"><a href="/warp/privacy/" class="nav-link">
  Privacy
</a></li></ul></li></ul></div></div> <a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/warp/documentation/Intro.html" class="nav-link">
  Documentation
</a></div><div class="nav-item"><a href="/warp/blog/" class="nav-link">
  Blog
</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="Learn More" class="dropdown-title"><span class="title">Learn More</span> <span class="arrow down"></span></button> <button type="button" aria-label="Learn More" class="mobile-dropdown-title"><span class="title">Learn More</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><h4>
          Explore
        </h4> <ul class="dropdown-subitem-wrapper"><li class="dropdown-subitem"><a href="/warp/research/" class="nav-link">
  Research
</a></li></ul></li><li class="dropdown-item"><h4>
          Resources
        </h4> <ul class="dropdown-subitem-wrapper"><li class="dropdown-subitem"><a href="/warp/contribution/" class="nav-link">
  Contribution Guide
</a></li><li class="dropdown-subitem"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer" class="nav-link external">
  Changelog
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></li><li class="dropdown-subitem"><a href="/warp/privacy/" class="nav-link">
  Privacy
</a></li></ul></li></ul></div></div> <a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></nav>  <ul class="sidebar-links"><li><section class="sidebar-group collapsable depth-0"><p class="sidebar-heading"><span>Welcome to WARP</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable depth-0"><p class="sidebar-heading"><span>About WARP</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group depth-0"><p class="sidebar-heading open"><span>Pipelines</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading"><span>CEMBA MethylC Seq Pipeline</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading"><span>Exome Germline Single Sample Pipeline</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading"><span>Genomic Data Commons Whole Genome Somatic Pipeline</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading"><span>Illumina Genotyping Arrays Pipeline</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading"><span>Optimus Pipeline</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading"><span>Single Cell ATAC Seq Pipeline</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading"><span>Smart-seq2 Multi Sample Pipeline</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading"><span>Smart-seq2 Single Sample Pipeline</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable is-sub-group depth-1"><p class="sidebar-heading open"><span>Whole Genome Germline Single Sample Pipeline</span> <span class="arrow down"></span></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/warp/documentation/Pipelines/Whole_Genome_Germline_Single_Sample_Pipeline/" aria-current="page" class="sidebar-link">Whole Genome Germline Single Sample Overview</a></li><li><a href="/warp/documentation/Pipelines/Whole_Genome_Germline_Single_Sample_Pipeline/wgs.methods.html" aria-current="page" class="active sidebar-link">Whole Genome Germline Single Sample v2.2.0 Methods</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/warp/documentation/Pipelines/Whole_Genome_Germline_Single_Sample_Pipeline/wgs.methods.html#detailed-methods" class="sidebar-link">Detailed Methods</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/warp/documentation/Pipelines/Whole_Genome_Germline_Single_Sample_Pipeline/wgs.methods.html#pre-processing-and-qc" class="sidebar-link">Pre-processing and QC</a></li><li class="sidebar-sub-header"><a href="/warp/documentation/Pipelines/Whole_Genome_Germline_Single_Sample_Pipeline/wgs.methods.html#variant-calling" class="sidebar-link">Variant Calling</a></li></ul></li></ul></li></ul></section></li></ul></section></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="whole-genome-germline-single-sample-v2-2-0-methods"><a href="#whole-genome-germline-single-sample-v2-2-0-methods" class="header-anchor">#</a> Whole Genome Germline Single Sample v2.2.0 Methods</h1> <p>The following contains a detailed methods description outlining the pipeline’s process, software, and tools that can be modified for a publication methods section.</p> <h2 id="detailed-methods"><a href="#detailed-methods" class="header-anchor">#</a> Detailed Methods</h2> <p>Preprocessing and variant calling was performed using the WholeGenomeGermlineSingleSample 2.2.0 pipeline using Picard 2.23.8, GATK 4.1.8, and Samtools 1.11 with default tool parameters unless otherwise specified. All reference files are available in the public <a href="https://console.cloud.google.com/storage/browser/gcp-public-data--broad-references/hg38/v0" target="_blank" rel="noopener noreferrer">Broad References Google Bucket<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>. The pipeline follows GATK Best Practices as previously described (<a href="https://www.oreilly.com/library/view/genomics-in-the/9781491975183/" target="_blank" rel="noopener noreferrer">Van der Auwera &amp; O'Connor, 2020<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>) as well as the Functional Equivalence specification (<a href="https://www.nature.com/articles/s41467-018-06159-4" target="_blank" rel="noopener noreferrer">Regier et al., 2018<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>).</p> <h3 id="pre-processing-and-qc"><a href="#pre-processing-and-qc" class="header-anchor">#</a> Pre-processing and QC</h3> <p>Whole genome paired-end reads in unmapped BAM (uBAM) format were first scattered to perform QC and alignment in parallel. Quality metrics were calculated using Picard CollectQualityYieldMetrics. uBAMs were converted to FASTQ using Picard SamToFastq and aligned to the Hg38 reference genome using BWA mem 0.7.15 with batch size set using -K 100000000. Metadata from the uBAMs was then merged with the aligned BAMs using Picard MergeBamAlignment with the parameters --SORT_ORDER=&quot;unsorted&quot;, allowing the data to be grouped by read name for efficient downstream marking of duplicates, and --UNMAP_CONTAMINANT_READS=true, to remove cross-species contamination.</p> <p>QC metrics (base distribution by cycle, insert size metrics, mean quality by cycle, and quality score distribution) were collected for the aligned, unsorted read-groups using Picard CollectMultipleMetrics. The read-group specific aligned BAMs were then aggregated and duplicate reads were flagged using Picard MarkDuplicates assuming queryname-sorted order and the parameter --OPTICAL_DUPLICATE_PIXEL_DISTANCE=2500, which is appropriate for patterned flowcells.</p> <p>The aggregated BAM file was then sorted using Picard SortSam with coordinate sort order. The fingerprints of separate read groups were verified using Picard CrosscheckFingerprints with a LOD threshold of -20. Cross-sample contamination was checked using verifyBamID2.</p> <p>The aligned BAM was then scattered for parallelization during base recalibration. A Base Quality Score Recalibration (BQSR) table was created with GATK BaseRecalibrator using the original base qualities (under the OQ Sam tag). The model was applied using ApplyBQSR with the static-quantized-quals option used according to the <a href="https://github.com/CCDG/Pipeline-Standardization/blob/master/PipelineStandard.md" target="_blank" rel="noopener noreferrer">Functional Equivalence specification<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> (<a href="https://www.nature.com/articles/s41467-018-06159-4" target="_blank" rel="noopener noreferrer">Regier et al., 2018<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>). Recalibrated BAM files were then merged using Picard GatherSortedBamFiles.</p> <p>QC metrics were calculated for the base-recalibrated BAM using Picard CollectMultipleMetrics. Fingerprints were verified using Picard CheckFingerprint and high duplication levels and chimerism were checked using the calculated summary metrics.</p> <p>To evaluate the coverage and performance of the whole genome sequencing experiment, the BAM was assessed using the Picard tools CollectWGSMetrics and CollectRawWgsMetrics.</p> <p>The final base-recalibrated BAM was converted to CRAM using Samtools view and validated using Picard ValidateSamFile.</p> <h3 id="variant-calling"><a href="#variant-calling" class="header-anchor">#</a> Variant Calling</h3> <p>Prior to variant calling, the variant calling interval list was subsetted to enable parallelization. Using the GATK PrintReads tool, the WellformedReadFilter was applied to reads. Variant calling was then applied to reads that passed the filter using GATK (v3.5) HaplotypeCaller with the parameters --max_alternate_alleles 3 (sufficient for human data),  --ERC GVCF, and --read_filter OverclippedRead (to reduce false positives resulting from contamination). The resulting GVCFs were merged using Picard MergeVcfs and the final VCF file was validated using GATK ValidateVariants. Variant metrics were calculated using Picard CollectVariantCallingMetrics.</p> <p>The pipeline’s final outputs included metrics, validation reports, an aligned CRAM with index, and a GVCF containing variant calls with an accompanying index.</p></div> <footer class="page-edit"><div class="edit-link"><a href="https://github.com/broadinstitute/warp/edit/master/website/docs/documentation/Pipelines/Whole_Genome_Germline_Single_Sample_Pipeline/wgs.methods.md" target="_blank" rel="noopener noreferrer">Edit this page on GitHub</a> <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></div> <div class="last-updated"><span class="prefix">Last Updated:</span> <span class="time">4/30/2021, 3:54:34 PM</span></div></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/warp/documentation/Pipelines/Whole_Genome_Germline_Single_Sample_Pipeline/" class="prev router-link-active">
        Whole Genome Germline Single Sample Overview
      </a></span> <!----></p></div> </main></div><div class="global-ui"><!----></div></div>
    <script src="/warp/assets/js/app.0913a268.js" defer></script><script src="/warp/assets/js/2.2043750e.js" defer></script><script src="/warp/assets/js/36.fcca5ed7.js" defer></script>
  </body>
</html>
